Training24/TicTacToe.c

```diff
@@ -0,0 +1,95 @@
+// TicTacToe.c
+// Program on Test3.2 branch.
+// TicTacToe.c: Implements a two-player Tic-Tac-Toe game.
+// ------------------------------------------------------------------------------------
+#pragma warning (disable:4996)
+#include <stdio.h>
+
+#define BOARD_SIZE 3
+#define WIN 1
+#define CONTINUE 0
+#define DRAW -1
+
+/// <summary>To print the TicTacToe game board.</summary>
+void PrintBoard (char board[BOARD_SIZE][BOARD_SIZE]) {
+   for (int i = 0; i < BOARD_SIZE; i++) {
+      for (int j = 0; j < BOARD_SIZE; j++) {
+         printf (" %c ", board[i][j]);
+         if (j < BOARD_SIZE - 1) printf ("|");
+      }
+      printf ("\n");
+      if (i < BOARD_SIZE - 1) printf ("---+---+---\n");
+   }
+   printf ("\n");
+}
+
+/// <summary>Checks for win, draw, or continue the game.</summary>
+int CheckGameStatus (char board[BOARD_SIZE][BOARD_SIZE], char player) {
+   for (int i = 0; i < BOARD_SIZE; i++) {
+      if ((board[i][0] == player && board[i][1] == player && board[i][2] == player) ||
+         (board[0][i] == player && board[1][i] == player && board[2][i] == player))
+         return WIN;
+   }
+   if ((board[0][0] == player && board[1][1] == player && board[2][2] == player) ||
+      (board[0][2] == player && board[1][1] == player && board[2][0] == player))
+      return WIN;
+   for (int i = 0; i < BOARD_SIZE; i++) {
+      for (int j = 0; j < BOARD_SIZE; j++) if (board[i][j] == ' ') return CONTINUE;
+   }
+   return DRAW;
+}
+
+/// <summary>Gets a valid move from the player.</summary> 
+int GetValidMove (char board[BOARD_SIZE][BOARD_SIZE], char player) {
+   char input[10];
+   int move;
+   while (1) {
+      printf ("\nPlayer %c, enter a move (1-9): ", player);
+      fgets (input, sizeof (input), stdin);
+      if (input[0] < '1' || input[0] > '9' || input[1] != '\0' && input[1] != '\n') {
+         printf ("Invalid input! Please enter a number between 1 and 9.\n");
+         continue;
+      }
+      move = input[0] - '0';
+      int row = (move - 1) / BOARD_SIZE,
+         col = (move - 1) % BOARD_SIZE;
+      if (board[row][col] == ' ') return move;
+      printf ("That spot is already taken! Please choose another spot.\n");
+   }
+}
+
+/// <summary>Starts two player TicTacToe game.</summary>
+void StartGame () {
+   char board[BOARD_SIZE][BOARD_SIZE] = { {' ', ' ', ' '}, {' ', ' ', ' '}, {' ', ' ', ' '} },
+      currentPlayer = 'X';
+   int move, gameStatus;
+   while (1) {
+      PrintBoard (board);
+      move = GetValidMove (board, currentPlayer);
+      int row = (move - 1) / BOARD_SIZE,
+         col = (move - 1) % BOARD_SIZE;
+      board[row][col] = currentPlayer;
+      gameStatus = CheckGameStatus (board, currentPlayer);
+      if (gameStatus == WIN) {
+         PrintBoard (board);
+         printf ("Player %c wins!\n", currentPlayer);
+         break;
+      }
+      if (gameStatus == DRAW) {
+         PrintBoard (board);
+         printf ("It's a draw!\n");
+         break;
+      }
```

Comment:
you can store both the board state and the messages in strings, and print them once at the end of the function